src/backup.rs

```diff
@@ -78,6 +78,60 @@ impl BackupEngine {
             Ok(())
         }
     }
+    /// Restore from the latest backup
+    /// db_dir:  input db directory
+    /// wal_dir:  input wal directory
+    /// opts: restore options
+    /// example:
+    /// use rocksdb::backup::{BackupEngine, BackupEngineOptions};
+    /// let backup_opts = BackupEngineOptions::default();
+    /// let backup_engine = BackupEngine::open(&backup_opts, &backup_path).unwrap();
+    /// let mut restore_option = rocksdb::backup::RestoreOptions::default();
+    /// restore_option.set_keep_log_files(true); /// true to keep log files
+    /// if let Err(e) = backup_engine.restore_from_latest_backup(&db_path, &wal_dir, &restore_option) {
+    ///     error!("Failed to restore from the backup. Error:{:?}", e);
+    ///     return Err(e.to_string());
+    ///  }
+    pub fn restore_from_latest_backup<P: AsRef<Path>>(
```

Comment:
Can you have two generic type arguments here, eg: `D: AsRef<Path>, W: AsRef<Path>`? That way each path can be provided by different types.